thespian/system/transport/TCPTransport.py

```diff
@@ -340,13 +340,12 @@ def getAdminAddr(capabilities):
     @staticmethod
     def getAddressFromString(addrspec, adminRouting=False):
         if isinstance(addrspec, tuple):
-            addrparts = addrspec
+            addrparts = list(addrspec)
         else:
             addrparts = addrspec.split(':')
         addrtype = RoutedTCPv4ActorAddress if adminRouting else TCPv4ActorAddress
-        if 1 == len(addrparts):
-            return ActorAddress(addrtype(addrparts[0], DEFAULT_ADMIN_PORT, external=True))
-        return ActorAddress(addrtype(addrparts[0], addrparts[1], external=True))
+        ip, port = addrparts[0], addrparts[1] if len(attrparts) > 1 else DEFAULT_ADMIN_PORT
+        return ActorAddress(addrtype(ip, port, external=True))
```

Comment:
I like this in general, but the ```ip, port = <ipval>, <complex statement>``` is potentially confusing.  It should also be ```len(addrparts)``` and not ```len(attrparts)``` -- mixed d's and t's.  In addition, since there's no additional usage of either, the variable assignment is unnecessary (although trivial) overhead.

What about:

   return ActorAddress(addrtype(addrparts[0],
                                                   addrparts[1] if len(addrparts) > 1 else DEFAULT_ADMIN_PORT,
                                                   external=True))